oak_functions/client/java/src/AttestationClient.java

```diff
@@ -99,17 +102,22 @@ protected void finalize() throws Throwable {
     /**
      * Encrypts and sends `message` via an attested gRPC channel to the server.
      */
-    public byte[] Send(byte[] message) throws Exception {
-        byte[] encryptedMessage = encryptor.encrypt(message);
-        AttestedInvokeRequest request = AttestedInvokeRequest
+    public Response Send(Request request) throws Exception {
```

Comment:
Why can't `Request` object be created inside `Send` ?
I think it's much simpler to have an API that accepts `byte[]`

Reply:
I think, the plan is to extend `Request`, and `SecureRequest` with additional fields, for example a Policy. In that case it is easier to work with `Request` instead of a byte array. I could send the binary encoded protobuf message, but then `send` would have to decode that to `Request` to construct `SecureRequest` from it.

This is because, these additional fields are going to be public (unencrypted). Otherwise, we could just work with the binary encoded protobuf.

Reply:
@ipetr0v I ended up using `Request` in more places, particularly in `experimental/grpc_attestation`. I believe when we have the generic state machine, all that will be moved into `oak_functions`.